gammapy/catalog/hawc.py

```diff
@@ -0,0 +1,134 @@
+    """
+
+    _source_name_key = 'source_name'
+    _source_index_key = 'catalog_row_index'
+
+    def info(self, info='all'):
+        """Print info.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'position', ' spectra'}
+            Comma separated list of options
+        """
+        ss = self.__str__(info=info)
+        print(ss)
+
+    def __str__(self, info='all'):
+        """Info string.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'position', 'spectra} #TODO: call spectra different
+            Comma separated list of options
+        """
+
+        if info == 'all':
+            info = 'basic,position,spectra'
+
+        ss = ''
+        ops = info.split(',')
+        if 'basic' in ops:
+            ss += self._info_basic()
+        if 'position' in ops:
+            ss += self._info_position()
+        if 'spectra' in ops:
+            ss += self._info_spectra() #TODO: call spectra different
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        ss = '\n*** Basic info ***\n\n'
+        ss += 'Catalog row index (zero-based) : {}\n'.format(d['catalog_row_index'])
+        ss += '{:<15s} : {}\n'.format('HWC2_FHL_name:', d['source_name'])
+        return ss
+
+    def _info_position(self):
+        """Print position info."""
+        d = self.data
+        ss = '\n*** Position info ***\n\n'
+
+        # TODO: Add here the SIMBAD position???
+
+        ss += 'Measurement:\n'
+        ss += '{:20s} : {:.3f}\n'.format('RA', d['ra'])
+        ss += '{:20s} : {:.3f}\n'.format('DEC', d['dec'])
+        ss += '{:20s} : {:.3f}\n'.format('GLON', d['glon'])
+        ss += '{:20s} : {:.3f}\n'.format('GLAT', d['glat'])
+        ss += '{:20s} : {:.3f}\n'.format('Position error', d['pos_err'])
+
+        return ss
+
+    def _info_spectra(self):
+        """Print spectral info."""
+        d = self.data
+        ss = '\n*** Spectral info ***\n\n'
+
+        ss += 'Spectrum 1:\n'
+        ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7TeV', d['spec0_dnde'].value, \
+            d['spec0_dnde_err'].value, d['spec0_dnde'].unit)
+        ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('index', d['spec0_index'], d['spec0_index_err'])
+        ss += '{:20s} : {:1}\n\n'.format('Test radius', d['spec0_radius'])
+
+        if(np.isnan(d['spec1_dnde'])):
+            ss += 'No second spectrum available for this source'
+        else:
+            ss += 'Spectrum 2:\n'
+            ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7TeV', d['spec1_dnde'].value, \
+                d['spec1_dnde_err'].value, d['spec1_dnde'].unit)
+            ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('index', d['spec1_index'], d['spec1_index_err'])
+            ss += '{:20s} : {:1}'.format('Test radius', d['spec1_radius'])
+        return ss
+
+    # TODO: Add other information to def ___str___(self, info)???
```

Comment:
Can you show the printout for one source in a comment in the PR?
That's much easier to review than the code.

Reply:
Currently it looks like this for the second source of the catalog:

*** Basic info ***

Catalog row index (zero-based) : 1
HWC2_FHL_name:  : 2HWC J0631+169

*** Position info ***

Measurement:
RA                   : 97.998 deg
DEC                  : 16.997 deg
GLON                 : 195.614 deg
GLAT                 : 3.507 deg
Position error       : 0.114 deg

*** Spectral info ***

Spectrum 1:
Flux at 7TeV         : 6.71e-15 +- 1.47e-15 1 / (cm2 s TeV)
index                : -2.570 +- 0.150
Test radius          : 0.0 deg

Spectrum 2:
Flux at 7TeV         : 4.87e-14 +- 6.85e-15 1 / (cm2 s TeV)
index                : -2.230 +- 0.080
Test radius          : 2.0 deg

Reply:
Looks mostly good. Some small suggestions:
- 'HWC2_FHL_name' -> 'Source name'
- '7TeV' -> '7 TeV' (insert space)
- 'index' -> 'Spectral index'
- '6.71e-15 +- 1.47e-15 1 / (cm2 s TeV)' -> '6.71e-15 +- 1.47e-15 cm-2 s-1 TeV-1' (write unit by hand; I find the number followed by '1 / ' hard to read and have seen others confused by it.

Reply:
Fixed.